client/extensions/woocommerce/app/order/order-payment/dialog.js

```diff
@@ -0,0 +1,226 @@
+		const { siteId } = this.props;
+
+		if ( siteId ) {
+			this.props.fetchPaymentMethods( siteId );
+		}
+	};
+
+	componentWillReceiveProps = newProps => {
+		const newSiteId = newProps.siteId;
+		const oldSiteId = this.props.siteId;
+
+		if ( oldSiteId !== newSiteId ) {
+			this.props.fetchPaymentMethods( newSiteId );
+		}
+	};
+
+	toggleDialog = () => {
+		this.setState( {
+			errorMessage: false,
+			refundTotal: 0,
+			refundNote: '',
+		} );
+		this.props.toggleDialog();
+	};
+
+	recalculateRefund = total => {
+		this.setState( { refundTotal: total } );
+	};
+
+	updateNote = event => {
+		this.setState( {
+			refundNote: event.target.value,
+		} );
+	};
+
+	sendRefund = () => {
+		const { order, paymentMethod, siteId, translate } = this.props;
+		const maxRefund = parseFloat( order.total ) + getOrderRefundTotal( order );
+		if ( this.state.refundTotal > maxRefund ) {
+			this.setState( {
+				errorMessage: translate( 'Refund must be less than or equal to the order total.' ),
+			} );
+			return;
+		} else if ( this.state.refundTotal <= 0 ) {
+			this.setState( { errorMessage: translate( 'Refund must be greater than zero.' ) } );
+			return;
+		}
+		this.toggleDialog();
+		const refundObj = {
+			amount: this.state.refundTotal + '', // API expects a string
+			reason: this.state.refundNote,
+			api_refund: paymentMethod && -1 !== paymentMethod.method_supports.indexOf( 'refunds' ),
+		};
+		this.props.sendRefund( siteId, order.id, refundObj );
+	};
+
+	renderCreditCard = () => {
+		const { isPaymentLoading, paymentMethod, translate } = this.props;
+		if ( isPaymentLoading ) {
+			return null;
+		}
+
+		if ( paymentMethod && -1 === paymentMethod.method_supports.indexOf( 'refunds' ) ) {
+			return (
+				<div className="order-payment__method">
+					<h3>{ translate( 'Manual Refund' ) }</h3>
+					<p>
+						{ translate(
+							"This payment method doesn't support automated refunds and must be submitted manually."
+						) }
+					</p>
+				</div>
+			);
+		}
+
+		return (
+			<div className="order-payment__method">
+				<h3>
+					{ translate( 'Refunding payment via %(method)s', {
+						args: {
+							method: paymentMethod.title,
+						},
+					} ) }
+				</h3>
+			</div>
+		);
+	};
+
+	render() {
+		const { isPaymentLoading, order, isVisible, translate } = this.props;
+		const { errorMessage, refundNote } = this.state;
+		const dialogClass = 'woocommerce'; // eslint/css specificity hack
+
+		if ( 'cancelled' === order.status || 'failed' === order.status ) {
+			return null;
+		}
+
+		let refundTotal = formatCurrency( 0, order.currency );
+		if ( this.state.refundTotal ) {
+			refundTotal = formatCurrency( this.state.refundTotal, order.currency );
+		}
+		refundTotal = refundTotal.replace( /[^0-9.,]/g, '' );
+
+		const dialogButtons = [
+			<Button onClick={ this.toggleDialog }>{ translate( 'Cancel' ) }</Button>,
+			<Button primary onClick={ this.sendRefund } disabled={ isPaymentLoading }>
+				{ translate( 'Refund' ) }
```

Comment:
Please add context for the translators. In this case, I think you want to ensure they use the _verb_ appropriate for their language.